projectroles/forms.py

```diff
@@ -541,9 +554,88 @@ def __init__(self, project=None, current_user=None, *args, **kwargs):
             self.fields['description'].widget = forms.HiddenInput()
             self.fields['readme'].widget = forms.HiddenInput()
 
+    @classmethod
+    def _validate_app_settings(
+        self,
+        cleaned_data,
+        app_plugins,
+        app_settings,
+        p_kwargs,
+        instance,
+        instance_owner_as,
+    ):
+        """Validate and clean app_settings form fields"""
+        errors = []
+
+        if cleaned_data.get('type') == PROJECT_TYPE_CATEGORY:
+            return (cleaned_data, errors)
+
+        for plugin in app_plugins + [None]:
+            if plugin:
+                name = plugin.name
+                p_settings = app_settings.get_definitions(
+                    APP_SETTING_SCOPE_PROJECT, app_name=name, **p_kwargs
+                )
+            else:
+                name = 'projectroles'
+                p_settings = app_settings.get_definitions(
+                    APP_SETTING_SCOPE_PROJECT, app_name=name, **p_kwargs
+                )
+
+            for s_key, s_val in p_settings.items():
+                s_field = 'settings.{}.{}'.format(name, s_key)
+
+                try:
+                    app_settings_errors = plugin.validate_app_settings(
+                        p_settings,
+                        s_key,
+                        project=instance,
+                        user=instance_owner_as,
+                    )
+                    if app_settings_errors:
+                        for field, error in app_settings_errors.items():
+                            if error:
+                                errors.append((field, error))
+                except AttributeError:
+                    # Plugin does not have a validate_app_settings method
+                    pass
+
+                if s_val['type'] == 'JSON':
+                    # for some reason, there is a distinct possibility, that the
+                    # initial value has been discarded and we get '' as value.
+                    # Seems to only happen in automated tests. Will catch that
+                    # here.
```

Comment:
I think you just fixed this in another PR by not raising `ValidationError`? Here we still seem to have the old code in.